geode-core/src/main/java/org/apache/geode/internal/tcp/Connection.java

```diff
@@ -3054,7 +3058,9 @@ private void readMessage(ByteBuffer peerDataBuffer) {
         failureMsg = "Unexpected failure deserializing message";
         failureEx = ex;
         rpId = md.getRPid();
-        logger.fatal("Unexpected failure deserializing message", failureEx);
+        // log at info level first in case fatal-level alert notification becomes blocked
+        logger.info(failureMsg, failureEx);
+        logger.fatal(failureMsg, failureEx);
```

Comment:
Instead of repeating this code in three places, comment-and-all, it would be better to have the three sites call a new method e.g. `logFatal(…)`. The method could have a block comment telling about what it's contract is etc.